packages/block-editor/src/hooks/metadata.js

```diff
@@ -29,39 +18,18 @@ export function addMetaAttribute( blockTypeSettings ) {
 		return blockTypeSettings;
 	}
 
-	const supportsBlockNaming = hasBlockMetadataSupport(
-		blockTypeSettings,
-		'name'
-	);
-
-	if ( supportsBlockNaming ) {
-		blockTypeSettings.attributes = {
-			...blockTypeSettings.attributes,
-			[ META_ATTRIBUTE_NAME ]: {
-				type: 'object',
-			},
-		};
-	}
+	blockTypeSettings.attributes = {
+		...blockTypeSettings.attributes,
+		[ META_ATTRIBUTE_NAME ]: {
+			type: 'object',
+		},
+	};
 
 	return blockTypeSettings;
 }
 
-export function addSaveProps( extraProps, blockType, attributes ) {
-	if ( hasBlockMetadataSupport( blockType ) ) {
-		extraProps[ META_ATTRIBUTE_NAME ] = attributes[ META_ATTRIBUTE_NAME ];
-	}
-
-	return extraProps;
-}
-
```

Comment:
This needed to be removed to allow tests to pass otherwise an additional `<div>` was added to output of all blocks.

All functionality and tests appears to work without it. I think it was mistakenly included in the original PR.